extensions/spring-xd-extension-gpload/src/main/java/org/springframework/xd/gpload/GploadTasklet.java

```diff
@@ -0,0 +1,294 @@
+
+	public void setHost(String host) {
+		this.host = host;
+	}
+
+	public int getPort() {
+		return port;
+	}
+
+	public void setPort(Integer port) {
+		this.port = port;
+	}
+
+	public String getUsername() {
+		return username;
+	}
+
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	public String getPasswordFile() {
+		return passwordFile;
+	}
+
+	public void setPasswordFile(String passwordFile) {
+		this.passwordFile = passwordFile;
+	}
+
+	@Override
+	protected boolean isStoppable() {
+		return false;
+	}
+
+	@Override
+	protected List<String> createCommand() throws Exception {
+		List<String> command = new ArrayList<String>();
+		command.add(getCommandString());
+		if (StringUtils.hasText(host)) {
+			command.add("-h");
+			command.add(host);
+		}
+		if (port != null) {
+			command.add("-p");
+			command.add(port.toString());
+		}
+		if (StringUtils.hasText(database)) {
+			command.add("-d");
+			command.add(database);
+		}
+		if (StringUtils.hasText(username)) {
+			command.add("-U");
+			command.add(username);
+		}
+		if (StringUtils.hasText(controlFile)) {
+			if (StringUtils.hasText(inputSourceFile)) {
+				try {
+					controlFileToUse = createControlFile(inputSourceFile, controlFile);
+				} catch (IOException e) {
+					throw new JobExecutionException("Error while creating control file", e);
+				}
+			}
+			else {
+				controlFileToUse = controlFile;
+			}
+			command.add("-f");
+			command.add(controlFileToUse);
+		}
+		if (StringUtils.hasText(options)) {
+			command.add(options);
+		}
+		return command;
+	}
+
+	@Override
+	protected String getCommandDisplayString() {
+		return getCommandString();
+	}
+
+	@Override
+	protected String getCommandName() {
+		return "gpload";
+	}
+
+	@Override
+	protected String getCommandDescription() {
+		return "gpload job";
+	}
+
+	@Override
+	public ExitStatus afterStep(StepExecution stepExecution) {
+		stepExecution.getExecutionContext().put(getCommandName() + ".controlFile", controlFileToUse);
+		return super.afterStep(stepExecution);
+	}
+
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		if (!StringUtils.hasText(gploadHome)) {
+			throw new IllegalArgumentException("Missing gploadHome property, it is mandatory for running gpload command");
+		}
+		if (StringUtils.hasText(password) && StringUtils.hasText(passwordFile)) {
+			throw new IllegalArgumentException("You can't specify both 'password' and 'passwordFile' options");
+		}
+		if (StringUtils.hasText(password)) {
+			addEnvironmentProvider(new GploadEnvironmentProvider(gploadHome, false, password));
+		}
+		else if (StringUtils.hasText(passwordFile)) {
+			addEnvironmentProvider(new GploadEnvironmentProvider(gploadHome, true, passwordFile));
+		}
+		else {
+			throw new IllegalArgumentException("You must specify either 'password' or 'passwordFile' option");
```

Comment:
The password handling is bad overall since there is no way to specify to never ever prompt for password. So the best compromise I could come up with was to force either password or passwordFile to avoid the majority of situations where we would get that password prompt.